Added genome assembly and patch versions. Fixes #1508.

ESapenaVentura opened the pull request:
### Release notes

For `type/file/analysis.json` schema:
- Added `genome_assembly_version` required field
- Added `genome_patch_version` field
 
### Why are these changes needed?

This change is needed for data consumers, as part of the metadata needed to correct for batch effects and harmonise the data to build proper cell atlases


### Reviews requested

- This is a major update, so it requires the review of all the DCP/2 approved PR reviewers

ESapenaVentura committed:
Added genome assembly and patch versions. Fixes #1508.

ESapenaVentura committed:
Fixed typo in new required field.

idazucchi commented on json_schema/type/file/analysis_file.json:
```diff
@@ -5,7 +5,8 @@
-        "file_core"
+        "file_core",
+        "genome_assembly_version"
     ],
```
```suggestion
    ],
    "dependentRequired": {
        "genome_patch_version": ["genome_assembly_version"]
    },
```

ESapenaVentura replied:
Love it, but I would put it with the `dependencies` keyword rather than with the `dependentRequired` (https://github.com/HumanCellAtlas/metadata-schema/pull/1473#discussion_r928916492)

Wkt8 replied:
Changed to dependencies! looks good.

Wkt8 committed:
Update json_schema/type/file/analysis_file.json

This commit will add a required dependency for 'genome_assembly_version' to be filled if 'genome_patch_version' is filled.

Co-authored-by: idazucchi <45209056+idazucchi@users.noreply.github.com>

Wkt8 reviewed (approved):
LGTM!

ESapenaVentura commented on json_schema/type/file/analysis_file.json:
```diff
@@ -5,8 +5,12 @@
+        "genome_assembly_version"
     ],
+    "dependencies": {
+        "genome_patch_version": ["genome_assembly_version"]
```
```suggestion
        "genome_patch_version": ["genome_assembly_version"]
        "genome_assembly_version": ["genome_patch_version"]
```

ESapenaVentura commented on json_schema/type/file/analysis_file.json:
```diff
@@ -5,8 +5,12 @@
+        "file_core",
+        "genome_assembly_version"
     ],
+    "dependencies": {
+        "genome_patch_version": ["genome_assembly_version"]
+    },
```
```suggestion
```

ESapenaVentura committed:
Deleted redundant dependency

ncalvanese1 reviewed (approved):
Looks good to me!

ESapenaVentura committed:
Ran release_prepare.py script.
